### ISSUE TITLE
Train Multi face with train_dreambooth.py on SD model

### ISSUE THREAD

**Issue by hosein-moayedi** [huggingface/diffusers#3175]
### Describe the bug

Hi there!
I have an issue with training the SD model with train_dreambooth.py.
I am doing these steps:

**Step 1:** 
Downloaded the Stable Diffusion from hugging face and using from train_dreambooth.py script to start to train my own face with some parameters and special SEED. 

**Step 2:**
After training, I saved the result as a .ckpt file on my local. 

**Step 3:**
Using that result model (.ckpt file that I saved in step 2) as a pre-trained model on my train_dreambooth.py parameter and give the dir of my friend's face to train that on my model with different SEED.

**Step 4:**
After finishing the training process I saved the new result as a .ckpt file and start using from that and give prompts to generate an image of faces.

**Problem:**
When I give the friend's name to generate an image from his face, I will see the right image of his face without any issues but when I try to write the prompt of generate an image from my face, I will see the generated image was 90% look like of my friend's face and 10% look like me!

**Note: "I defined my face name as a special uuid version 4 and my friend's face has a different uuid"**

I think the last train is overwriting on my previous train!! 

How can I fix this issue?

### Reproduction

**this first train params:**

!accelerate launch train_dreambooth.py \
  --pretrained_model_name_or_path="runwayml/stable-diffusion-v1-5" \
  --pretrained_vae_name_or_path="stabilityai/sd-vae-ft-mse" \
  --output_dir=$OUTPUT_DIR \
  --revision="fp16" \
  --with_prior_preservation --prior_loss_weight=0.1 \
  --seed=1337 \ ---->>>> **Different SEED**
  --resolution=512 \
  --train_batch_size=1 \
  --train_text_encoder \
  --mixed_precision="fp16" \
  --use_8bit_adam \
  --gradient_accumulation_steps=1 \
  --learning_rate=1e-7 \
  --lr_scheduler="constant" \
  --lr_warmup_steps=80 \
  --num_class_images=120 \
  --sample_batch_size=4 \
  --max_train_steps=800 \
  --save_interval=10000 \
  --save_sample_prompt="05b62e68-8513-4779-9309-5643f27f288a" \
  --concepts_list="concepts_list.json"

**The second training params:**

!accelerate launch train_dreambooth.py \
  --pretrained_model_name_or_path="path of .ckpt result from last train." \
  --pretrained_vae_name_or_path="stabilityai/sd-vae-ft-mse" \
  --output_dir=$OUTPUT_DIR \
  --revision="fp16" \
  --with_prior_preservation --prior_loss_weight=0.1 \
  --seed=9385 \. ---->>>> **Different SEED**
  --resolution=512 \
  --train_batch_size=1 \
  --train_text_encoder \
  --mixed_precision="fp16" \
  --use_8bit_adam \
  --gradient_accumulation_steps=1 \
  --learning_rate=1e-7 \
  --lr_scheduler="constant" \
  --lr_warmup_steps=80 \
  --num_class_images=120 \
  --sample_batch_size=4 \
  --max_train_steps=800 \
  --save_interval=10000 \
  --save_sample_prompt="05b62e68-8513-4779-9309-5643f27f288a" \
  --concepts_list="concepts_list.json"

### System Info

Please check this: https://github.com/ShivamShrirao/diffusers/tree/main/examples/dreambooth

**Comment by innat-asj** [comment]
@hosein-moayedi 
Sequential fine-tuning of a model lost information (if not totally). Instead, train the Lora model (with dream booth) for each subject.

**Comment by hosein-moayedi** [comment]
Thanks @innat-asj 
Your way is very cool.
but I have some problem with lora
please help me more 🙏

When I start to train my model with this script, everything is ok with these parameters:
https://github.com/ShivamShrirao/diffusers/blob/main/examples/dreambooth/train_dreambooth.py

!accelerate launch train_dreambooth.py
--pretrained_model_name_or_path="runwayml/stable-diffusion-v1-5"
--pretrained_vae_name_or_path="stabilityai/sd-vae-ft-mse"
--output_dir=$OUTPUT_DIR
--revision="fp16"
--with_prior_preservation --prior_loss_weight=0.1
--seed=1337 \ ---->>>> Different SEED
--resolution=512
--train_batch_size=1
--train_text_encoder
--mixed_precision="fp16"
--use_8bit_adam
--gradient_accumulation_steps=1
--learning_rate=1e-7
--lr_scheduler="constant"
--lr_warmup_steps=80
--num_class_images=120
--sample_batch_size=4
--max_train_steps=800
--save_interval=10000
--save_sample_prompt="05b62e68-8513-4779-9309-5643f27f288a"
--concepts_list="concepts_list.json"


but when I start to train my model using this: https://github.com/ShivamShrirao/diffusers/blob/main/examples/dreambooth/train_dreambooth_lora.py
with common parameters, in the training process, the value of the loss has a lot of changes between 0.3 to 0.03 in each training step!! however in "train_dreambooth.py" script "lost" is very stable and that will change between 0.150 and 0.145 and stop in 0.145.

what is the problem?!

also after train with lora with dreambooth, I am getting black result!!

**Comment by innat-asj** [comment]
I also noticed unstable loss values but ignored that anyway as I got desired output after training. For the black image, I think you can remove the safety checker.  Like as follows 

```python
def get_base_model(model_base):
    pipe = StableDiffusionPipeline.from_pretrained(
        model_base, torch_dtype=torch.float16,
        safety_checker=None,
    )
    return pipe
```

Here is one of my trails. By default the lora rank is set 4 and size of the weight file would be roughly 3 MB.

```bash
accelerate launch dreambooth_lora.py \
  --pretrained_model_name_or_path=$MODEL_NAME  \
  --instance_data_dir=$INSTANCE_DIR \
  --output_dir=$OUTPUT_DIR \
  --instance_prompt="a photo of sks dog"  \
  --with_prior_preservation \
  --prior_loss_weight=2.0 \
  --class_data_dir=$CLASS_DIR \
  --class_prompt="a photo of dog" \
  --num_class_images=70 \  # I have pre-saved 70 samples for prior prev.
  --resolution=512 \
  --train_batch_size=1 \
  --gradient_accumulation_steps=1 \
  --gradient_checkpointing \
  --enable_xformers_memory_efficient_attention \
  --use_8bit_adam \
  --learning_rate=5e-6 \ # see the mentioned blog post down below.
  --lr_scheduler="constant" \
  --lr_warmup_steps=0 \
  --max_train_steps=2000 \ # in some test, try: 5000/more.
  --seed="0" \
```

[Good reading.](https://huggingface.co/blog/dreambooth)

**Comment by patrickvonplaten** [comment]
Also cc @sayakpaul @williamberman here

**Comment by williamberman** [comment]
@hosein-moayedi this is the expected behavior of dreambooth, the base script, at least in its current form does not support multiple subjects in the same model

**Comment by sayakpaul** [comment]
Multi-concept training is still challenging with the current DreamBooth script. Would suggest looking into Custom Diffusion: https://github.com/huggingface/diffusers/tree/main/examples/custom_diffusion.

It has got good support for multi-concept training: https://github.com/huggingface/diffusers/tree/main/examples/custom_diffusion#training-on-multiple-concepts-

**Comment by hosein-moayedi** [comment]
> Multi-concept training is still challenging with the current DreamBooth script. Would suggest looking into Custom Diffusion: https://github.com/huggingface/diffusers/tree/main/examples/custom_diffusion.
> 
> It has got good support for multi-concept training: https://github.com/huggingface/diffusers/tree/main/examples/custom_diffusion#training-on-multiple-concepts-

Hi
thanks for replay
Is it working for train model sequentially? ex: you train a face of person 1 on model and save that and retrain that model with face of other person (2) without damage to first train (person 1)??

and will have damage to pertain base model after train some faces?

**Comment by sayakpaul** [comment]
This is something we have not explored yet. Since this issue relates to more of a technical discussion and is not really a bug in the codebase, I would suggest posting it on either our [Discord forum](https://hf.co/join/discord) or the [Discussion forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63). 

Hence, I am closing this issue for now.

**Comment by hosein-moayedi** [comment]
Hello!
I tried to use from train_dreambooth_lora but after finishing training, I can't see an image of my face that I trained!

**This is my training parameters:**

```
MODEL_NAME="XpucT/Deliberate"
INSTANCE_DIR="datasets/instance-images/1fdfdf359622486c94e742e3e1a975cb/"
OUTPUT_DIR="output-models/1fdfdf359622486c94e742e3e1a975cb/"
CLASS_DIR="datasets/class-man/"

!accelerate launch train_dreambooth_lora.py \
  --pretrained_model_name_or_path=$MODEL_NAME  \
  --instance_data_dir=$INSTANCE_DIR \
  --class_data_dir=$CLASS_DIR \
  --output_dir=$OUTPUT_DIR \
  --instance_prompt="<hosein>" \
  --class_prompt="photo of a man" \
  --resolution=512 \
  --train_batch_size=1 \
  --gradient_accumulation_steps=1 \
  --checkpointing_steps=100 \
  --with_prior_preservation \
  --prior_loss_weight=0.1 \
  --learning_rate=1e-7 \
  --lr_scheduler="constant" \
  --lr_warmup_steps=80 \
  --num_class_images=120 \
  --max_train_steps=1000 \
  --seed="9274" \
  --mixed_precision="fp16" \
  --train_text_encoder \
  --use_8bit_adam \
  --enable_xformers_memory_efficient_attention
```


**Now make a pipeline and apply Lora weights:**

```
import torch
import random
from torch import autocast
from diffusers import StableDiffusionPipeline, DDIMScheduler, DPMSolverMultistepScheduler
from IPython.display import display

model_path = "output-models/1fdfdf359622486c94e742e3e1a975cb/"


def get_base_model(model_base):
    pipe = StableDiffusionPipeline.from_pretrained(model_base, torch_dtype=torch.float16, safety_checker=None)
    pipe.scheduler = DPMSolverMultistepScheduler.from_config(pipe.scheduler.config)
    pipe.to("cuda")
    return pipe

def assign_lora_weight(pipe, lora_weight_id):
    pipe.unet.load_attn_procs(lora_weight_id)
    return pipe

# get base model for one time
base_pipe = get_base_model("XpucT/Deliberate")

# lora-layer-painter-wise.
hosein_pip = assign_lora_weight(base_pipe, lora_weight_id=model_path)


```


**Now I want make new image from my face:**

```
image = hosein_pip(    
    "<hosein>",
    height=512,
    width=512,
    negative_prompt="blender, cropped, lowres, poorly drawn face, out of frame, poorly drawn hands, blurry, bad art, blurred, text, watermark, disfigured, blurry background, deformed, closed eyes, animation, CGI ",
    num_inference_steps=25,
    guidance_scale=7.5,
    cross_attention_kwargs={"scale": 2.5},
).images[0]

image.save(f"result/{random.randint(1,1000000)}.jpg")
```


**After making an image, that is a very random instance like a Car or train or other human face that is not me...

what is the problem here?!**

**Comment by innat-asj** [comment]
```
  --instance_prompt="a photo of hsn person" \
  --class_prompt="a photo of a person" 
```
